Added cylFF.py and 9-Far-field-irradiance-and-NA-fiber.ipynb.

matt8s opened the pull request:
I have added code, in cylFF.py, to calculate the far field emission pattern from LP modes. In 9-Far-field-irradiance-and-NA-fiber.ipynb, I have demonstrated its use, provided an alternative method for calculating a V-b curve, and used the far field pattern to get the emission angular spread vs core radius. I think these additions may be helpful to users of the code.

matt8s committed:
Created cylFF.py and 9-Far-field-irradiance-and-NA-fiber.ipynb.

Commit:
Restored b[0] calc using LP_mode_value to Far-field ipynb.

Commit:
Increased sweep resolution.

Commit:
Added degree unit to final plot legend, to fix typo.

scottprahl commented:
Thanks for your contribution!  Hopefully I will have time to review this over the weekend.

scottprahl commented:
I integrated your functions into the cylinder.step.py file.  I improved solving for b for large V.  I also made the calculation for b work when an array of V is passed.

I revised your Far Field notebook and added the calculation of a few more graphs from Chen. I removed the extra division by 2 that you were doing and now the result matches the graph in Chen more closely.

Finally, if I teach this class again, I might use Chen's book.  It seems better than Ghatak's.

matt8s commented:
Hi Prof. Prahl, I finally had the chance to review your corrections and improvements. Thank you for incorporating my work into your project, and for doing that work! Sincerely, Matthew Spotnitz, PhD Sandia National Laboratories On Feb 13, 2024, at 4:55 PM, Scott Prahl ***@***.***> wrote: I integrated your functions into the cylinder.step.py file. I improved solving for b for large V. I also made the calculation for b work when an array of V is passed. I revised your Far Field notebook and added the calculation of a few more graphs from Chen. I removed the extra division by 2 that you were doing and now the result matches the graph in Chen more closely. Finally, if I teach this class again, I might use Chen's book. It seems better than Ghatak's. — Reply to this email directly, view it on GitHub , or unsubscribe . You are receiving this because you authored the thread. Message ID: ***@***.***>
